keps/sig-network/20190125-ingress-api-group.md

```diff
@@ -555,6 +558,117 @@ backend:
     name: my-bucket
 ```
 
+### Status change for controllers like external-dns
+
+There is a logical bug, that hopefully no one had, with Ingress `status`
+and [external-dns](https://github.com/kubernetes-sigs/external-dns) if multiple hostnames are defined in one ingress.
+
+#### Background how external-dns works with ingress
+
+Example how external-dns works:
+
+```
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name:my-ingress
+spec:
+  rules:
+  - host: myapp.example.com
+    http:
+      paths:
+      - backend:
+          serviceName: myapp
+          servicePort: 80
+status:
+  loadBalancer:
+    ingress:
+    - hostname: my-load-balancer1.target.example.cloud
+```
+
+In the above example, external-dns would setup a DNS CNAME record,
+that points from `myapp.example.com` to
+`my-load-balancer1.target.example.cloud`.
+
+This idea works reliably for us since 4 years, but there is a
+theoretical problem, which I would like to address in ingress GA.
+
+#### Problem - Ingress data
+
+The problem is, that there is nothing that prevents from having many
```

Comment:
Interesting proposition.  There's an assumption (unstated, AFAICT) that 1 Ingress object == 1 LB instance, and that host matching is done by that LB instance.  In the case of multiple status.ingress, external-dns should set up multiple A/AAAA records for the same name (and for the CNAME case - that's an error).

You're saying (if I understand) that we should allow a single Ingress object to map N host rules to M ingress points.  Is that right?

We don't generally design to hypotheticals - do we have any examples of implementations that would want to do this if it was allowed?  Or customer use-cases that demand this?  Why would they not simply create multiple Ingress objects?

Assuming we wanted to support it, we'd also have to consider it in the new Gateway API design - @bowei @rramkumar1

Reply:
@thockin makes sense to me, too. Maybe the right fix is to add it to the ingress documentation!

Reply:
@szuecs -- are you going to send the doc fix? is the use of multiple ingress objects instead meet your use case?

Reply:
It's fine for me and our use case. 
I was only worried about the unclear spec and wanted to show it with an example.
I also have to check what external-dns does in this case.

Reply:
I checked and external-dns does the right thing.